pkg/security/ptracer/process.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package ptracer holds the start command of CWS injector
+package ptracer
+
+import "github.com/DataDog/datadog-agent/pkg/security/proto/ebpfless"
+
+// Resources defines shared process resources
+type Resources struct {
+	Fd  map[int32]string
+	Cwd string
+}
+
+// Process represents a process context
+type Process struct {
+	Pid  int
+	Tgid int
+	Nr   map[int]*ebpfless.SyscallMsg
+	Res  *Resources
+}
+
+// NewProcess returns a new process
+func NewProcess(pid int) *Process {
+	return &Process{
+		Pid:  pid,
+		Tgid: pid,
+		Nr:   make(map[int]*ebpfless.SyscallMsg),
+		Res: &Resources{
+			Fd: make(map[int32]string),
+		},
+	}
+}
+
+// ProcessCache defines a thread cache
+type ProcessCache struct {
+	pid2Process map[int]*Process
+	tgid2Pid    map[int]map[int]bool
+}
+
+// NewProcessCache returns a new thread cache
+func NewProcessCache() *ProcessCache {
+	return &ProcessCache{
+		pid2Process: make(map[int]*Process),
+		tgid2Pid:    make(map[int]map[int]bool),
```

Comment:
I would rather use a map[int][]int here, I don't see the utility of this bool value

Reply:
Indeed, will do